internal/nexodus/exit_node.go

```diff
@@ -0,0 +1,234 @@
+package nexodus
+
+import (
+	"fmt"
+	"github.com/nexodus-io/nexodus/internal/util"
+)
+
+const (
+	oobDNS           = 53
+	oobHttps         = 443
+	oobGoogleStun    = 19302
+	wgFwMark         = "51820"
+	oobFwMark        = "19302"
+	oobFwdMarkHex    = "0x4B66"
+	nfExitNodeTable  = "nexodus-exit-node"
+	nfOobMangleTable = "nexodus-oob-mangle"
+	nfOobSnatTable   = "nexodus-oob-snat"
+)
+
+// ExitNodeClientSetup setups up the routing tables, netfilter tables and out of band connections for the exit node client
+func (nx *Nexodus) ExitNodeClientSetup() error {
+	nx.exitNode.exitNodeClientEnabled = true
+
+	// Lock deviceCache for read
+	nx.deviceCacheLock.RLock()
+	defer nx.deviceCacheLock.RUnlock()
+
+	for _, deviceEntry := range nx.deviceCache {
+		for _, allowedIp := range deviceEntry.device.AllowedIps {
+			if util.IsDefaultIPv4Route(allowedIp) {
+				// assign the device advertising a default route as the exit node server/origin node
+				nx.exitNode.exitNodeOrigins[0] = wgPeerConfig{
+					PublicKey:           deviceEntry.device.PublicKey,
+					Endpoint:            deviceEntry.device.EndpointLocalAddressIp4,
+					AllowedIPs:          deviceEntry.device.AllowedIps,
+					PersistentKeepAlive: "25",
```

Comment:
See https://www.wireguard.com/quickstart/ 

```
This is called persistent keepalives. When this option is enabled, a keepalive packet is sent to the server endpoint once every interval seconds. A sensible interval that works with a wide variety of firewalls is 25 seconds. Setting it to 0 turns the feature off, which is the default, since most users will not need this, and it makes WireGuard slightly more chatty. This feature may be specified by adding the PersistentKeepalive = field to a peer in the configuration file, or setting persistent-keepalive at the command line. If you don't need this feature, don't enable it. But if you're behind NAT or a firewall and you want to receive incoming connections long after network traffic has gone silent, this option will keep the "connection" open in the eyes of NAT.
```

Reply:
in `wg_deploy.go` we have:

```
const (
	persistentKeepalive = "20"
)
```

We should use a consistent default. You can just use that one. Change it to 25 if you'd like.

Reply:
Ty!